File: test-register.js
// Test script for registration
const axios = require('axios');

const API_URL = 'http://localhost:8000/api';

async function testRegistration() {
  const userData = {
    // Let backend use email as username
    email: 'testuser123@example.com',
    password: 'Password123!',
    password2: 'Password123!',
    first_name: 'Test',
    last_name: 'User',
    role: 'student',
    user_type: 'student',
    student_id: 'TEST123456',
    department: 'PGDCA',
    year_of_study: 1,
    phone_number: '1234567890'
  };

  console.log('Attempting to register with data:', userData);
  
  try {
    // Define all possible registration endpoints
    const endpoints = [
      { url: `${API_URL}/auth/register/`, name: 'API Auth Register' },
      { url: `${API_URL}/accounts/register/`, name: 'Accounts Register' },
      { url: `${API_URL.replace('/api', '')}/accounts/register/`, name: 'Root Accounts Register' }
    ];
    
    // Try each endpoint in sequence
    for (const endpoint of endpoints) {
      try {
        console.log(`Trying registration endpoint: ${endpoint.name} (${endpoint.url})`);
        const response = await axios.post(endpoint.url, userData, {
          headers: {
            'Content-Type': 'application/json'
          },
          timeout: 10000
        });
        
        console.log(`Registration successful with ${endpoint.name}:`);
        console.log(JSON.stringify(response.data, null, 2));
        return;
      } catch (error) {
        console.error(`Error with ${endpoint.name}:`);
        if (error.response) {
          console.error('Response status:', error.response.status);
          console.error('Response data:', JSON.stringify(error.response.data, null, 2));
        } else if (error.request) {
          console.error('No response received:', error.message);
        } else {
          console.error('Error:', error.message);
        }
        
        // Only try the next endpoint if it was a connection error
        if (!error.response || error.response.status >= 500) {
          continue;
        } else {
          // For validation errors (400), don't try other endpoints
          break;
        }
      }
    }
    
    console.log('All registration attempts failed');
  } catch (error) {
    console.error('General error:', error.message);
  }
}

testRegistration(); 

File: test_register.py
import json
import requests

# Base URL for the API
BASE_URL = 'http://localhost:8000'

def test_registration():
    """Test user registration"""
    # Data for non-student user (to avoid student-specific validations)
    user_data = {
        'email': 'tester@example.com',
        'password': 'Test123!',
        'password2': 'Test123!',
        'first_name': 'Test',
        'last_name': 'User',
        'user_type': 'teacher',  # Using teacher instead of student to avoid extra validation
    }
    
    # Try the accounts registration endpoint directly
    print("\n=== Trying /accounts/register/ endpoint ===")
    response = requests.post(f"{BASE_URL}/accounts/register/", json=user_data)
    print(f"Status: {response.status_code}")
    try:
        print(f"Response: {json.dumps(response.json(), indent=2)}")
    except:
        print(f"Raw response: {response.text}")
    
    # Try auth registration endpoint
    print("\n=== Trying /api/auth/register/ endpoint ===")
    response = requests.post(f"{BASE_URL}/api/auth/register/", json=user_data)
    print(f"Status: {response.status_code}")
    try:
        print(f"Response: {json.dumps(response.json(), indent=2)}")
    except:
        print(f"Raw response: {response.text}")

def test_login(email, password):
    """Test user login"""
    login_data = {
        'email': email,
        'password': password
    }
    
    # Try accounts login endpoint
    print("\n=== Trying /accounts/login/ endpoint ===")
    response = requests.post(f"{BASE_URL}/accounts/login/", json=login_data)
    print(f"Status: {response.status_code}")
    try:
        print(f"Response: {json.dumps(response.json(), indent=2)}")
    except:
        print(f"Raw response: {response.text}")
    
    # Try auth login endpoint
    print("\n=== Trying /api/auth/login/ endpoint ===")
    response = requests.post(f"{BASE_URL}/api/auth/login/", json=login_data)
    print(f"Status: {response.status_code}")
    try:
        print(f"Response: {json.dumps(response.json(), indent=2)}")
    except:
        print(f"Raw response: {response.text}")

if __name__ == "__main__":
    # First try with existing users
    print("=== Testing login with existing user ===")
    test_login('student@example.com', 'student123')
    
    # Then try registration
    print("\n=== Testing new user registration ===")
    test_registration()
    
    # Finally try login with the new user
    print("\n=== Testing login with new user ===")
    test_login('tester@example.com', 'Test123!') 